src/camera_utils.py

```diff
@@ -45,24 +45,30 @@ def project_world_point_to_image(camera: Camera, point: np.ndarray) -> np.ndarra
 
 def compute_image_footprint_on_surface(camera: Camera, distance_from_surface: float) -> np.ndarray:
     """Compute the footprint of the image captured by the camera at a given distance from the surface.
-
+    
+    Footprint = (image size ) * (distance) / (focal length)
+    
     Args:
         camera (Camera): the camera model.
         distance_from_surface (float): distance from the surface (in m).
 
     Returns:
         np.ndarray: [footprint_x, footprint_y] in meters.
     """
-    raise NotImplementedError()
+    footprint_x = (camera.image_size_x_px * distance_from_surface) / camera.fx
+    footprint_y = (camera.image_size_y_px * distance_from_surface) / camera.fy
+    return np.array([footprint_x, footprint_y])
+
 
 def compute_ground_sampling_distance(camera: Camera, distance_from_surface: float) -> float:
     """Compute the ground sampling distance (GSD) at a given distance from the surface.
-
+    
     Args:
         camera (Camera): the camera model.
         distance_from_surface (float): distance from the surface (in m).
     
     Returns:
         float: the GSD in meters (smaller among x and y directions).
     """
-    raise NotImplementedError()
+    focal_length_max = max(camera.fx, camera.fy)
+    return ((distance_from_surface) / focal_length_max)  
```

Comment:
Nit: the parantheses are not needed. You can simply write `return distance_from_surface / focal_length_max`